### PR TITLE
Add signee status toggle in users list

### PR DESCRIPTION
**Pull request by rluodev**
Also added link to org documents in deletion request card.

This closes issues #4699 and #4694.

Screenshot 2023-12-11 at 10 48 27 PM

image

### CODE REVIEW

File: app/views/events/_organization_position_invite.html.erb [hunk 1]
```diff
@@ -27,10 +27,17 @@
           <span>No contract signature</span>
```
Comment (sampoder) [inline app/views/events/_organization_position_invite.html.erb]:
```suggestion
          <span>No contract signature</span>
          <%= link_to "(toggle)", toggle_signee_status_organizer_path(position.id), data: { turbo_method: "post", turbo_frame: "_top" } %>
```
thoughts on moving the toggle up here? feels weird to have two separate items in the list related to contract signature.

File: app/views/events/_organization_position_invite.html.erb [hunk 2]
```diff
@@ -27,10 +27,17 @@
...
         <%= inline_icon "leader", size: 20, style: "vertical-align: middle; margin: -0.2rem 0.5rem 0 0;" %>
         <%= link_to "Impersonate this user", impersonate_user_path(position.user.id, return_to: request.url), data: { turbo_method: "post", turbo_frame: "_top" } %>
       </div>
+      <div>
+        <% if position.signee? %>
+          <%= inline_icon "checkmark", size: 20, style: "vertical-align: middle; margin: -0.2rem 0.5rem 0 0;" %>
+        <% else %>
+          <%= inline_icon "checkbox", size: 20, style: "vertical-align: middle; margin: -0.2rem 0.5rem 0 0;" %>
+        <% end %>
+        <%= link_to "Toggle contract signature status", toggle_signee_status_organizer_path(position.id), data: { turbo_method: "post", turbo_frame: "_top" } %>
```
Comment (sampoder) [inline app/views/events/_organization_position_invite.html.erb]:
```suggestion
      <div>
        <% if position.signee? %>
          <%= inline_icon "checkmark", size: 20, style: "vertical-align: middle; margin: -0.2rem 0.5rem 0 0;" %>
        <% else %>
          <%= inline_icon "checkbox", size: 20, style: "vertical-align: middle; margin: -0.2rem 0.5rem 0 0;" %>
        <% end %>
        <%= link_to "Toggle contract signature status", toggle_signee_status_organizer_path(position.id), data: { turbo_method: "post", turbo_frame: "_top" } %>
      </div>
```
closing tag :D

**Review by sampoder — CHANGES_REQUESTED** [review]
tested locally, works like a charm! a couple of quick comments from me

**Comment by sarthaktexas** [comment]
Two things since you're already working on the card:
1. Could we fix the UI of the organizer being deleted? Just make it more clear who's being deleted.

2. Could we add a direct download link to the Fiscal Sponsorship Agreement?

**Review by sarthaktexas — CHANGES_REQUESTED** [review]
Above comments.

**Comment by hackliv** [comment]
I assume this would be a feature going forward since it might take too much time to edit that for every existing user currently. Can we create another issue where when someone requests a user deletion they have to check a box indicating whether the user is a contract signee?

**Review by hackliv — APPROVED** [review]

**Comment by rluodev** [comment]
> Two things since you're already working on the card:
> 
> 1. Could we fix the UI of the organizer being deleted? Just make it more clear who's being deleted.
> 2. Could we add a direct download link to the Fiscal Sponsorship Agreement?

1 is done, not sure how to do 2 tbh

**Comment by rluodev** [comment]
also I think one of you needs to review as hcb-engineers

File: app/controllers/organizer_positions_controller.rb [hunk 3]
```diff
@@ -58,4 +58,15 @@ def mark_visited
...
   end
 
+  def toggle_signee_status
+    organizer_position = OrganizerPosition.find(params[:id])
+    authorize organizer_position
+    if organizer_position.signee?
+      organizer_position.update!(is_signee: false)
+    else
+      organizer_position.update!(is_signee: true)
+    end
```
Comment (cjdenio) [inline app/controllers/organizer_positions_controller.rb]:
```suggestion
    organizer_position.toggle!(:is_signee)
```

File: config/routes.rb [hunk 4]
```diff
@@ -234,10 +234,11 @@
     post "cancel"
   end
 
-  resources :organizer_positions, only: [:destroy], as: "organizers" do
+  resources :organizer_positions, only: [:destroy, :toggle_signee_status], as: "organizers" do
```
Comment (cjdenio) [inline config/routes.rb]:
```suggestion
  resources :organizer_positions, only: [:destroy], as: "organizers" do
```

**Comment by sarthaktexas** [comment]
@Soar1826 In the future, always keep the top-most comment updated.
This is what the admin side looks like now:
image

I also get this error on testing, presumably from Caleb's change in `routes.rb`:
image

**Review by sarthaktexas — CHANGES_REQUESTED** [review]
Above.

**Comment by rluodev** [comment]
My laptop is dead and I’m editing code on my phone on GitHub. @sarthaktexas can you check if that new commit fixed it?

**Comment by cjdenio** [comment]
@sarthaktexas you'll need to have Turbo enabled in your dev environment for the link to work (since it relies on `data-turbo-method`). (http://localhost:3000/my/settings/previews)

**Comment by rluodev** [comment]
@sarthaktexas @cjdenio @sampoder This should be good to go.

**Comment by sarthaktexas** [comment]
Hey! I can't approve this without trying it first so give me tomorrow to figure out how to enable Turbo. Great work!

**Review by sampoder — APPROVED** [review]
Code looks good to me - thanks for the great work on this Ruien!

File: app/views/events/_organization_position_invite.html.erb [hunk 5]
```diff
@@ -26,8 +26,9 @@
           <%= inline_icon "docs", size: 20, style: "vertical-align: middle; margin: -0.2rem 0.5rem 0 0;" %>
           <span>No contract signature</span>
         <% end %>
+        <%= link_to "(toggle)", toggle_signee_status_organizer_path(position.id), data: { turbo_method: "post", turbo_frame: "_top" } %>
```
Comment (sarthaktexas) [inline app/views/events/_organization_position_invite.html.erb]: Will this create a problem if an admin is not on Turbo? Or are all admins on Turbo?

**Review by sarthaktexas — APPROVED** [review]
Just like Sam said, works like a charm once Turbo is on! Great work :)

File: app/views/events/_organization_position_invite.html.erb (hunk 5 above)
Comment (sarthaktexas) [reply app/views/events/_organization_position_invite.html.erb]: This might be a question for @cjdenio

Comment (rluodev) [reply app/views/events/_organization_position_invite.html.erb]: Turbo is a feature enabled for everyone on prod now.

File: app/views/events/_organization_position_invite.html.erb [hunk 6]
```diff
@@ -30,8 +30,9 @@
...
```
Comment (garyhtou) [inline app/views/events/_organization_position_invite.html.erb]: Let's make it a bit harder to accidentally toggle this

```suggestion
        <%= link_to "(toggle)", toggle_signee_status_organizer_path(position.id), data: { turbo_method: "post", turbo_frame: "_top", turbo_confirm: "Are you sure you want to toggle #{position.user.name}'s contract signee status?" } %>
```

**Review by garyhtou — APPROVED** [review]

**Pull request merged**